### PR TITLE
[_6465] allow remote(...) hints to target INST_NAME

### PR DESCRIPTION
**Pull request by d-w-moore**
Replaces https://github.com/irods/irods/pull/6467

### CODE REVIEW

**Comment by trel** [comment]
can `getTaggedValues` throw?

**Comment by d-w-moore** [comment]
Yes indeed, if the argument is not well-formed. thanks.

**Comment by trel** [comment]
right, looks good to me - please touch up the clang-formatting and await alan/kory eyes.

**Comment by korydraughn** [comment]
The logic looks good.

Are you planning on adding a test? That should be doable using the topology settings and metadata.

**Comment by d-w-moore** [comment]
@korydraughn  I'm definitely interested in writing the test and was going to ask about how to do it. You mention the topology settings and metadata, is there a guide or example? Thanks!

**Comment by d-w-moore** [comment]
@trel Not sure what's meant by the clang formatting

File: plugins/rule_engines/irods_rule_language/src/functions.cpp
Comment (alanking) [inline plugins/rule_engines/irods_rule_language/src/functions.cpp:1912]: "instance_name" -> `irods::KW_CFG_INSTANCE_NAME`

**Review by alanking — COMMENTED** [review]
Overall looks good. I agree with clang-format and test sentiments. As @korydraughn suggested, if you use the `irods.test.settings.HOSTNAME_X` variables in your `remote()` call's target host(s), it can run in and outside of topology tests without having to selectively skip the tests and exercise your code.

File: plugins/rule_engines/irods_rule_language/src/functions.cpp
Comment (d-w-moore) [reply plugins/rule_engines/irods_rule_language/src/functions.cpp:1912]: noticed that, will use the symbol.

Comment (d-w-moore) [reply plugins/rule_engines/irods_rule_language/src/functions.cpp]: hm, there are several keywords that would all seem to fit, but the simplest seems to be INSTANCE_NAME_KW just as it is here, in a function called from _delayExec: https://github.com/irods/irods/blob/fc0653187ec7a308a466529aa76917c723c06995/server/re/src/irods_re_structs.cpp#L519

Comment (alanking) [reply plugins/rule_engines/irods_rule_language/src/functions.cpp]: Oh, I didn't know that existed, or that we had multiple of that keyword. I think what you used is better, personally. Let's go with `INSTANCE_NAME_KW`.

**Comment by alanking** [comment]
One more clang-format thing and I think this will be good to go

**Comment by d-w-moore** [comment]
> One more clang-format thing and I think this will be good to go

If i understand correctly, do i need to add some clang format directives ?

**Comment by alanking** [comment]
You can configure `clang-format` with `git` by following the instructions in the GitHub Actions workflow: https://github.com/irods/irods/blob/fc0653187ec7a308a466529aa76917c723c06995/.github/workflows/linter-irods-clang-format.yml#L29-L35

*Or* you can apply the diff directly using `git apply` since it's pretty small:
```bash
$ echo '
>  index c1c7b2d8f..4b4c9bf77 100644
> --- a/plugins/rule_engines/irods_rule_language/src/functions.cpp
> +++ b/plugins/rule_engines/irods_rule_language/src/functions.cpp
> @@ -1908,8 +1908,8 @@ Res *smsi_remoteExec( Node** paramsr, int, Node* node, ruleExecInfo_t* rei, int,
>      try {
>          auto taggedValues = getTaggedValues(params[1]->text);
>          auto it = taggedValues.find("INST_NAME");
> -        if ( it != taggedValues.end() ) {
> -            addKeyVal( &execMyRuleInp.condInput, INSTANCE_NAME_KW, it->second.front().c_str());
> +        if (it != taggedValues.end()) {
> +            addKeyVal(&execMyRuleInp.condInput, INSTANCE_NAME_KW, it->second.front().c_str());
>              taggedValues.erase(it);
>          }
>          it = taggedValues.find("ZONE");
> ' > /tmp/applyme
$ git apply /tmp/applyme
```

**OR** you can just delete the two spaces that it wants you to delete. :p

**Review by alanking — APPROVED** [review]
Overall seems good to me. Shall we squash? Leave off the # in case other thoughts come in

**Comment by trel** [comment]
let’s squash em, no #

**Comment by d-w-moore** [comment]
Done!

**Comment by korydraughn** [comment]
I didn't see any topology-based tests in the PR.

Did we conclude they aren't necessary?

**Comment by alanking** [comment]
I think we determined that that's going to be hard because we would have to configure remote servers from inside the test and/or read from a remote log (see https://github.com/irods/irods/issues/4164).

**Comment by trel** [comment]
Right, this was tested by hand as I understand.

**Comment by alanking** [comment]
Got verbal approval, so let's go ahead and add #'s.

**Comment by d-w-moore** [comment]
Will do. I assume that goes for the irods_docs (https://github.com/irods/irods_docs/pull/164) and Python RE  plugin (https://github.com/irods/irods_rule_engine_plugin_python/pull/128) PRs against `main` as well ?

**Comment by alanking** [comment]
No, just this one. We can address the other PRs in turn.

**Comment by alanking** [comment]
If you feel that they must all be merged together, then hold off on #'s until all of them are approved.

**Comment by d-w-moore** [comment]
coolio. done....

**Pull request merged**
